File: app/src/main/java/yangFenTuoZi/runner/plus/utils/ExceptionUtils.java
package yangFenTuoZi.runner.plus.utils;

import android.app.Activity;
import android.content.Context;
import android.os.Looper;

import com.google.android.material.dialog.MaterialAlertDialogBuilder;

import yangFenTuoZi.runner.plus.R;
import yangFenTuoZi.runner.plus.server.Logger;

public class ExceptionUtils {

    public static void throwableToDialog(Activity context, Throwable tr) {
        throwableToDialog(context, getStackTraceString(tr));
    }

    public static void throwableToDialog(Activity context, String errorMsg) {
        if (Looper.getMainLooper() == Looper.myLooper())
            createDialog(context, errorMsg);
        else
            context.runOnUiThread(() -> createDialog(context, errorMsg));
    }

    private static void createDialog(Context context, String errorMsg) {
        new MaterialAlertDialogBuilder(context)
                .setTitle(R.string.error)
                .setMessage(errorMsg)
                .show();
    }

    public static String getStackTraceString(Throwable tr) {
        return Logger.getStackTraceString(tr);
    }
}

File: app/src/main/java/yangFenTuoZi/runner/plus/utils/ExceptionUtils.kt
package yangFenTuoZi.runner.plus.utils

import android.app.Activity
import android.content.Context
import android.os.Looper
import android.util.Log.getStackTraceString
import com.google.android.material.dialog.MaterialAlertDialogBuilder
import yangFenTuoZi.runner.plus.R

object ExceptionUtils {

    fun Throwable.toErrorDialog(context: Activity) {
        getStackTraceString(this).toErrorDialog(context)
    }

    fun CharSequence.toErrorDialog(context: Activity) {
        if (Looper.getMainLooper() == Looper.myLooper()) {
            createDialog(context, this)
        } else {
            context.runOnUiThread { createDialog(context, this) }
        }
    }

    private fun createDialog(context: Context, errorMsg: CharSequence) {
        MaterialAlertDialogBuilder(context)
            .setTitle(R.string.error)
            .setMessage(errorMsg)
            .show()
    }
}